app/models/spotlight/reindexing_log_entry.rb

```diff
@@ -0,0 +1,42 @@
+module Spotlight
+  ##
+  # a log entry representing an attempt to reindex some number of records in an exhibit
+  class ReindexingLogEntry < ActiveRecord::Base
+    enum job_status: { unstarted: 0, in_progress: 1, succeeded: 2, failed: 3 }
+
+    belongs_to :exhibit, class_name: 'Spotlight::Exhibit'
+    belongs_to :user, class_name: '::User'
+
+    # null start times sort to the top, to more easily surface pending reindexing
+    default_scope { order('start_time IS NOT NULL, start_time DESC') }
+    scope :recent, -> { limit(5) }
+
+    def duration
+      end_time - start_time if end_time
+    end
+
+    def reindexing_started!
+      self.start_time = Time.zone.now
+      self.job_status = :in_progress
+      save
+    rescue
+      Rails.logger.error "unexpected error updating log entry to :in_progress from #{caller}"
+    end
+
+    def reindexing_succeeded!
+      self.end_time = Time.zone.now
+      self.job_status = :succeeded
```

Comment:
ah, i was unaware of that clever rails helper.  i'm having a little trouble finding documentation on a `succeeded!` method in active record or rails in general... could you provide a link or elaborate on usage, @cbeer?

i guess if we were worried about the `succeeded!` method auto-saving and callers being unaware, we could at least comment the method.  it seems like if a developer saw it, and was already familiar with the rails method, they'd know that it saved, and if they were unfamiliar with the method, they'd hopefully look at the code and comments to see how it works.  plus, the `!` convention signals some state change in the object itself.